lib/jellyfish/component/hls.ex

```diff
@@ -20,34 +22,37 @@ defmodule Jellyfish.Component.HLS do
 
   @impl true
   def config(options) do
-    storage = fn directory -> %LLStorage{directory: directory, room_id: options.room_id} end
-    RequestHandler.start(options.room_id)
-
-    {:ok,
-     %HLS{
-       rtc_engine: options.engine_pid,
-       owner: self(),
-       output_directory: output_dir(options.room_id),
-       mixer_config: %MixerConfig{
-         video: %CompositorConfig{
-           stream_format: %Membrane.RawVideo{
-             width: 1920,
-             height: 1080,
-             pixel_format: :I420,
-             framerate: {24, 1},
-             aligned: true
+    with {:ok, valid_opts} <- OpenApiSpex.cast_value(options, ApiSpec.Component.HLS.schema()) do
+      hls_storage = setup_hls_storage(options.room_id, low_latency?: valid_opts.lowLatency)
+
+      {:ok,
+       %HLS{
+         rtc_engine: options.engine_pid,
+         owner: self(),
+         output_directory: output_dir(options.room_id),
+         mixer_config: %MixerConfig{
+           video: %CompositorConfig{
+             stream_format: %Membrane.RawVideo{
+               width: 1920,
+               height: 1080,
+               pixel_format: :I420,
+               framerate: {24, 1},
+               aligned: true
+             }
            }
+         },
+         hls_config: %HLSConfig{
+           hls_mode: :muxed_av,
+           mode: :live,
+           target_window_duration: :infinity,
+           segment_duration: @segment_duration,
+           partial_segment_duration: @partial_segment_duration,
```

Comment:
right now in both scenarios, SinkBin will create partial segments.
```suggestion
           partial_segment_duration: (if validopts.lowLatency, do: @partial_segment_duration, else: nil),
```

Reply:
😵‍💫